src/blueapi/service/interface.py

```diff
@@ -0,0 +1,205 @@
+from collections.abc import Mapping
+from typing import Any
+
+from blueapi.config import ApplicationConfig
+from blueapi.core.context import BlueskyContext
+from blueapi.core.event import EventStream
+from blueapi.messaging.base import MessagingTemplate
+from blueapi.messaging.stomptemplate import StompMessagingTemplate
+from blueapi.service.model import (
+    DeviceModel,
+    PlanModel,
+    WorkerTask,
+)
+from blueapi.worker.event import TaskStatusEnum, WorkerState
+from blueapi.worker.reworker import TaskWorker
+from blueapi.worker.task import Task
+from blueapi.worker.worker import TrackableTask, Worker
+
+"""This module provides interface between web application and underlying Bluesky
+context and worker"""
+
+
+class _Singleton:
+    context: BlueskyContext
+    worker: Worker
+    messaging_template: MessagingTemplate | None = None
+    initialized = False
```

Comment:
I was experimenting to see if we could make our earlier idea of using `lru_cached` functions work, see here https://github.com/DiamondLightSource/blueapi/pull/557

I think this fixes the testing issue by clearing the caches on teardown?

Reply:
I think it will help yes, have merged.